### PR TITLE
Feature/connecting search

### PR DESCRIPTION
**Pull request by alina26**

### CODE REVIEW

**Review by RKDGT — APPROVED** [review]

File: src/components/apartaments/Apartments.jsx [hunk 1]
```diff
@@ -1,13 +1,162 @@
...
+  useEffect(() => {
+    setHotelsList([])
+    setCounter(-1)
+    if (props.valueFromParent !== undefined) {
+      setCatchProps(props.valueFromParent.data_from_child)
+    } else {
+      setCatchProps(props)
+    }
+  }, [props])
+  console.log(catchProps)
```
Comment (IraCherginyshka) [inline src/components/apartaments/Apartments.jsx]: delete console

File: src/components/apartaments/Apartments.jsx [hunk 2]
```diff
@@ -1,13 +1,162 @@
...
+  useEffect(() => {
+    console.log(catchProps.name)
```
Comment (IraCherginyshka) [inline src/components/apartaments/Apartments.jsx]: delete

File: src/components/apartaments/Apartments.jsx [hunk 3]
```diff
@@ -1,13 +1,162 @@
...
+            overview: data.data.data.body.overview.overviewSections[0].content.join(),
+            address: data.data.data.body.propertyDescription.address.fullAddress,
+            starRating: data.data.data.body.propertyDescription.starRating,
+            guestRating: data.data.data.body.guestReviews.formattedRating
+          };
+          setHotel(newHotel);
+        })
+        .catch((error) => {
+          console.error(error);
+          setIsSearchSuccesful(false);
```
Comment (IraCherginyshka) [inline src/components/apartaments/Apartments.jsx:68]: error, setError ???

File: src/components/apartaments/Apartments.jsx [hunk 4]
```diff
@@ -1,13 +1,162 @@
...
+            newHotel = {
+              ...hotel,
+              pictUrl: data.data.hotelImages[0].baseUrl.replace('{size}', data.data.hotelImages[0].sizes[0].suffix)
+            };
+          }
+          if (hotelsBaseInfo[counter] !== undefined) {
+            if (hotelsBaseInfo.length < 5) {
+              if (counter < hotelsBaseInfo.length) {
+                setHotelsList([...hotelsList, newHotel]);
+                setCounter(counter + 1);
```
Comment (IraCherginyshka) [inline src/components/apartaments/Apartments.jsx:89]: I don't like counter

File: src/components/apartaments/Apartments.jsx (hunk 1 above)
Comment (RKDGT) [reply src/components/apartaments/Apartments.jsx]: oops.. ok:)

File: src/components/search-panel/SearchPanel.jsx [hunk 5]
```diff
@@ -28,15 +27,35 @@ const Styles = styled.div`
...
     return (<div>
         <input placeholder={props.placeholder}
-            className='searchPanel' />
+            className='searchPanel'
+            onChange={city => getValueFromSearch(city.target.value)} />
     </div>
     );
 };
 
+function getStart(st) {
```
Comment (IraCherginyshka) [inline src/components/search-panel/SearchPanel.jsx:49]: What does this function do?

File: src/components/apartaments/Apartments.jsx (hunk 3 above)
Comment (RKDGT) [reply src/components/apartaments/Apartments.jsx:68]: this setState is used to check ....    to show the user that the search was successful or uncompleted ... cause if the user enters incorrect data, the api will return an error

File: src/components/search-panel/SearchPanel.jsx (hunk 5 above)
Comment (RKDGT) [reply src/components/search-panel/SearchPanel.jsx:49]: callback func to get date from "checkIn". We will rename it

File: src/components/apartaments/Apartments.jsx (hunk 3 above)
Comment (IraCherginyshka) [reply src/components/apartaments/Apartments.jsx:68]: So you can setError and display "Something went wrong" for example
it is up to you

File: src/components/search-panel/SearchPanel.jsx (hunk 5 above)
Comment (IraCherginyshka) [reply src/components/search-panel/SearchPanel.jsx:49]: weird function

**Review by IraCherginyshka — APPROVED** [review]

**Pull request merged**